packages/qdrant-loader-core/src/qdrant_loader_core/llm/settings.py

```
@@ -25,6 +25,33 @@ class RateLimitPolicy:
+    if isinstance(value, str):
+        text = value.strip()
+        if not text:
+            return None
+        try:
+            return int(text)
+        except ValueError:
+            return None
+    return None
+
+
+def _to_int_with_default(value: Any, default: int) -> int:
```

Ruff (0.15.18)

[warning] Dynamically typed expressions (typing.Any) are disallowed in `value`

(ANN401)